### PR TITLE
Snapshot/restore/bdd

### PR DESCRIPTION
**Pull request by tiagolobocastro**
test(snapshot/bdd): add initial restore delete
    
    A bug was found where after snapshot delete the parent volume needs to have
    ts allocation cache deleted.
    ( The WA for now is to restart the io-engine container and force update )
    
    todo: add multiple layers of snapshots and restores
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    fix(snapshot/volume): split allocation counters
    
    Splits allocation counters into replicas, snapshots and both.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    fix(snapshot/pool): add specific error when pool cannot be used
    
    todo: enhance filtering of resources to include reasons why a certain
    resource could not be used.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

### CODE REVIEW

File: control-plane/stor-port/src/types/v0/transport/volume.rs [hunk 1]
```diff
@@ -76,23 +76,41 @@ pub struct VolumeState {
...
+    /// Allocated size in bytes, related the largest healthy replica, including snapshots.
     /// For example, if a volume has 2 replicas, each with 1MiB allocated space, then
     /// this field will be 1MiB.
     allocated: u64,
-    /// Allocated size in bytes, accrued from all the replica.
-    /// For example, if a volume has 2 replicas, each with 1MiB allocated space, then
-    /// this field will be 2MiB.
+    /// Allocated size in bytes, related the largest healthy replica, excluding snapshots.
+    allocated_replica: u64,
+    /// Allocated size in bytes, related the largest healthy replica's snapshots.
```
Comment (Abhinandan-Purkait) [inline control-plane/stor-port/src/types/v0/transport/volume.rs]: Is it largest healthy replica's snapshot or largest snapshot of a that replica?

**Review by Abhinandan-Purkait — APPROVED** [review]

File: control-plane/stor-port/src/types/v0/transport/volume.rs (hunk 1 above)
Comment (tiagolobocastro) [reply control-plane/stor-port/src/types/v0/transport/volume.rs]: In code it's the snapshot of the largest healthy replica, but maybe should be the largest snapshot among all healthy replica, wdyt?

Comment (Abhinandan-Purkait) [reply control-plane/stor-port/src/types/v0/transport/volume.rs]: Yeah, not sure. The current one is fine probably?

Comment (tiagolobocastro) [reply control-plane/stor-port/src/types/v0/transport/volume.rs]: I've split largest replica and largest snapshot.

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**